libs/src/services/creatorNode/CreatorNodeSelection.ts

```diff
@@ -274,41 +386,56 @@ class CreatorNodeSelection extends ServiceSelection {
         )
-        const { storagePathSize, storagePathUsed, maxStorageUsedPercent } = resp.response.data.data
+        const { storagePathSize, storagePathUsed, maxStorageUsedPercent } =
+          resp.response.data.data
         if (maxStorageUsedPercent) {
           this.maxStorageUsedPercent = maxStorageUsedPercent
         } else {
-          console.warn(`maxStorageUsedPercent not found in health check response. Using constructor value of ${this.maxStorageUsedPercent}% as maxStorageUsedPercent.`)
+          console.warn(
+            `maxStorageUsedPercent not found in health check response. Using constructor value of ${this.maxStorageUsedPercent}% as maxStorageUsedPercent.`
+          )
         }
-        const hasEnoughStorage = this._hasEnoughStorageSpace({ storagePathSize, storagePathUsed })
+        const hasEnoughStorage = this._hasEnoughStorageSpace(
+          storagePathSize,
+          storagePathUsed
+        )
         isHealthy = isUp && versionIsUpToDate && hasEnoughStorage
       }
 
-      if (!isHealthy) { this.addUnhealthy(endpoint) }
+      if (!isHealthy) {
+        this.addUnhealthy(endpoint)
+      }
 
       return isHealthy
     })
 
     // Create a mapping of healthy services and their responses. Used on dapp to display the healthy services for selection
     // Also update services to be healthy services
-    const servicesMap = {}
-    const healthyServicesList = healthyServices.map(service => {
-      servicesMap[service.request.id] = service.response.data
-      return service.request.id
+    const servicesMap: Record<string, AxiosResponse['data']> = {}
+    const healthyServicesList = healthyServices.map((service) => {
+      const requestId = service.request.id as string
+      servicesMap[requestId] = service.response?.data
+      return service.request.id as string
     })
 
-    this.decisionTree.push({ stage: DECISION_TREE_STATE.FILTER_OUT_UNHEALTHY_OUTDATED_AND_NO_STORAGE_SPACE, val: healthyServicesList })
+    this.decisionTree.push({
+      stage:
+        DECISION_TREE_STATE.FILTER_OUT_UNHEALTHY_OUTDATED_AND_NO_STORAGE_SPACE,
+      val: healthyServicesList
+    })
 
     // Record metrics
-    if (this.creatorNode && this.creatorNode.monitoringCallbacks.healthCheck) {
-      healthCheckedServices.forEach(check => {
-        if (check.response && check.response.data) {
+    if (this.creatorNode?.monitoringCallbacks.healthCheck) {
+      healthCheckedServices.forEach((check) => {
+        if (check.response?.data) {
           const url = new URL(check.request.url)
           const data = check.response.data.data
           try {
+            // @ts-expect-error we make a check that it exists above, not sure why this isn't caught
             this.creatorNode.monitoringCallbacks.healthCheck({
               endpoint: url.origin,
               pathname: url.pathname,
+              // @ts-expect-error TODO: this is probably a bug since URL doesn't contain queryrString
```

Comment:
super nit: typo `queryString`

Reply:
Should probably be `searchParams`?

Reply:
that was intentionally mispelled :D and yeah maybe we change to searchParams? what's the move here you think?

Reply:
lol totally missed that double typo 😂  

I don't think it could hurt since it's clearly broken :shrug:

Reply:
the original code had "queryrString" :O

would reach out to the file authors here and see about correcting while we're at it